website/addons/wiki/utils.py

```diff
@@ -154,3 +155,64 @@ def format_wiki_version(version, num_versions, allow_preview):
         raise InvalidVersionError
 
     return version
+
+def serialize_wiki_settings(user, node_ids):
```

Comment:
It's also worth noting that `include_wiki_settings` recurses through all the descendants, which may slow this down even further.